cli/crud/import/containers.go

```diff
@@ -0,0 +1,163 @@
+	CrudCtx crudCtxСontainer `yaml:"crudCtx" json:"crudCtx"`
+}
+
+// batchСontainer contains the context of the batch, including necessary meta information for crud.
+type batchСontainer struct {
+	// BatchNumber is batch number during import work.
+	BatchNumber uint32 `yaml:"batchNumber" json:"batchNumber"`
+	// Header сontains a header of input data.
+	Header []string `yaml:"header" json:"header"`
+	// TuplesAmount сontains the number of initialized tuples within batch.
+	TuplesAmount uint32 `yaml:"tuplesAmount" json:"tuplesAmount"`
+	// Tuple is slice of tuples within this batch.
+	Tuple []batchTupleСontainer `yaml:"tuples" json:"tuples"`
+}
+
+// progressCtxСontainer contains the context for the progress file.
+type progressCtxСontainer struct {
+	// StartTimestamp contains the work start time of this launch.
+	StartTimestamp string `yaml:"startTimestamp" json:"startTimestamp"`
+	// LastDumpTimestamp contains the time of last write to progress file of this launch.
+	LastDumpTimestamp string `yaml:"lastDumpTimestamp" json:"lastDumpTimestamp"`
+	// EndOfFileReached indicates has the END been reached or not.
+	EndOfFileReached bool `yaml:"endOfFileReached" json:"endOfFileReached"`
+	// LastPosition contains last processing position of input file in this launch.
+	LastPosition uint32 `yaml:"lastPosition" json:"lastPosition"`
+	// RetryPosition contains positions of input file that could not be imported in this launch.
+	RetryPosition []uint32 `yaml:"retryPositions" json:"retryPositions"`
+	// lastPositionPrevProgress contains LastPosition from previous version of progress file.
+	lastPositionPrevProgress uint32
+	// retryPositionPrevProgress contains RetryPosition from previous version of progress file.
+	retryPositionPrevProgress []uint32
+}
+
+// batchSequenceCtxСontainer contains the context for
+// organizing movement of batch window on input file.
+type batchSequenceCtxСontainer struct {
+	// batchSize is tuples amount within batch.
+	batchSize uint32
+	// batchCounter is batch counter during import work (used to fill batchСontainer->BatchNumber).
+	batchCounter uint32
+	// tupleNumber is contains the number of the tuple within the current batch.
+	// Used to fill batchTupleСontainer->TupleNumber.
+	tupleNumber uint32
+	// header сontains a header of input data.
+	header []string
+}
+
+// unparsedCsvReaderCtxСontainer contains private fields for organizing context of
+// raw reading of input data for import.
+// Raw reading is necessary to get a string of data in an unparsed form.
+// The standard parser does not allow this.
+// It also contains fields for implementing a mechanism to eliminate
+// the inconsistency of the position counter due to a multi-line CSV records.
+// Fields inside this structure should not be read or overwritten by the programmer.
+// Exclusion are methods updCtxAfterParseError, updCtxAfterRecordHandling.
+type unparsedCsvReaderCtxСontainer struct {
+	_masterPosition uint32
```

Comment:
In golang, fields starting with lowercase letters are private, do not use underscores here.